tests/h/search/connection_test.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import pytest
+
+from elasticsearch_dsl.connections import connections
+from elasticsearch_dsl import Index
+
+from h import search
+from h.search.connection import connect
+
+
+def remove_connection():
+    """Remove elasticsearch connection made in test"""
+    connections.remove_connection('foobar')
+
+
+class TestConnection(object):
+    def test_connect_is_available_in_search_api(self):
+        assert connect == search.connect
+
+    def test_connect_creates_default_connection(self):
+        # search.connect is invoked as part of test bootstrapping
+        # so the connection should already be established
+        assert connections.get_connection()
+        assert connections.get_connection('default')
+        assert connections.get_connection() == connections.get_connection('default')
+
+    def test_connect_allows_additional_aliases(self, request):
+        request.addfinalizer(remove_connection)
+
+        connections.create_connection(alias='foobar')
+
+        assert connections.get_connection('foobar')
+
+    def test_connect_does_not_raise_on_invalid_host(self, request):
```

Comment:
This test is here (perhaps/likely temporarily as it's not actually testing the behavior of `connect` itself) to raise an important point for discussion. When a new ES connection is established—whether the "old" version via `Elasticsearch` or in the "new" version via `elasticsearch_dsl` (which is a thin wrapper over `Elasticsearch` when it comes to connections)—`Elasticsearch` _will not raise_ if the connection is no good. e.g. pass it a nonsensical `host` and it will instantiate happily in both cases. It's not until the connection is subsequently _used_ (queried, examined, etc.) that you'll get an Exception if the connection is bad.

I lost about two hours to this today as I was making myself crazy trying to get `elasticsearch_dsl.connections.create_connection` to raise. I'm not having a lot of luck with Google fu today to establish what best practices are around "checking to see if a given ES connection is working." There's the notion of _sniffing_, but that feels like a rabbit hole to some extent (FWIW: setting a `sniff_on_start` keyword arg to `True` when creating a new connection _will_ raise). I feel like there is something on a conceptual level here that I'm not entirely grasping so I'm backing away from this for the now. Perhaps someone else on the team with more actual ES experience can advise.

Reply:
When discussing with Sean he recommended just querying a stat end point thereby using the connection without actually adding an index or anything. I think that’s a good approach.